docs/adr/0004-instrumentation.md

```diff
@@ -0,0 +1,100 @@
+# Implement instrumentation via a new API Endpoint with external ad hoc probe calculation
+
+- Status: proposed
+- Deciders: Christina Harlow, Linh Nguyen, Rob Miller
+- Date: 2022-04-05
+
+Technical Story: https://github.com/mozilla/glam/issues/1847
+
+## Context and Problem Statement
+
+In order to help us with various product decisions and overall product quality,
+we want to understand how GLAM is normally used: Where does it fit in a typical
+user's workflow (a starting point, an endpoint, or something in between)?
+
+We can accomplish this by gathering data such as how a typical user session
+looks like, most and least looked at probes, how probe browsing behaves over
+short and long periods of time and whether it correlates with other events such
+as certain product releases.
+
+Additionally, as a more time-sensitive issue, GLAM ETL does a lot of work and
+takes over 12 hours to complete. We intend to reduce GLAM ETL time by finding
+the probes that are most likely to be requested by users from the aforementioned
+data and have the ETL build only those probes. This document also presents a
+plan for ad hoc probe building, which needs to happen when someone requests a
+probe that had not been pre-built by the ETL.
+
+## Decision Drivers
+
+- Usage metrics collection must be reliable and cannot be bypassed or influenced
+  by other systems such as browsers' adblocks
+- GLAM ETL should consume this service so it can optimize its work, only
+  processing probes that are likely to be looked at
+- GLAM App should consume this service so it can trigger an ad hoc calculation
+  whenever a probe that was not processed by GLAM ETL is requested
```

Comment:
Interesting. Thanks for the heads up. 
With the current state of things I think the first time such probe gets requested it will have to go through the ad hoc processing, then we can flag it as "built" for the day and check that flag for subsequent requests.

Reply:
Are you planning to have a state table with entry for each probe, and date last processed? Perhaps used by a different api endpoint